### PR TITLE
Make sync and async disposal of Renderer behave the same

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Components/Components/src/RenderTree/Renderer.cs [hunk 1]
```diff
@@ -956,10 +955,31 @@ private void HandleExceptionViaErrorBoundary(Exception error, ComponentState? er
         /// Releases all resources currently used by this <see cref="Renderer"/> instance.
         /// </summary>
         /// <param name="disposing"><see langword="true"/> if this method is being invoked by <see cref="IDisposable.Dispose"/>, otherwise <see langword="false"/>.</param>
-        protected virtual void Dispose(bool disposing)
+        protected virtual void Dispose(bool disposing) => DisposeAsync().AsTask().GetAwaiter().GetResult();
```
Comment (pranavkm) [inline src/Components/Components/src/RenderTree/Renderer.cs:958]: @javiercn you have the most context here - do you remember why `Dispose` doesn't implement by calling `DisposeAsync` and blocking?

Comment (javiercn) [reply src/Components/Components/src/RenderTree/Renderer.cs:958]: Likely because that was virtual and would break consumers

Comment (SteveSandersonMS) [reply src/Components/Components/src/RenderTree/Renderer.cs]: Blocking is impossible and throws on WebAssembly. I know you don't normally dispose a renderer on WebAssembly but nothing stops you.

**Review by SteveSandersonMS — APPROVED** [review]
Seems this doesn't change the sync-vs-async disposal thing any more, so if it's just moving the `HotReloadManager.OnDeltaApplied -= ...` instruction, looks very straightforwards :)

**Comment by pranavkm** [comment]
Yeah, the virtual bit is an important detail we wouldn't to needlessly change.

**Pull request merged**